_posts/2016/11/2016-11-21-gap-lesson.md

```diff
@@ -0,0 +1,151 @@
+schools in computational discrete mathematics](http://www.codima.ac.uk/schools/),
+which are primarily intended for PhD students and researchers from UK
+institutions. A typical school starts with the Software Carpentry workshop
+covering basic concepts and tools, such as working with the command line,
+version control and task automation, continued with introductions to GAP
+and SageMath systems, and followed by the series of lectures and exercise
+classes on a selection of topics in computational discrete mathematics.
+
+This naturally led to the idea of establishing a Software Carpentry lesson
+on programming with GAP. I started to develop it in 2015 for our
+[first training school in Manchester](http://www.codima.ac.uk/school2015/).
+Since I have never been at any of the Software Carpentry workshops before and
+had not yet completed instructor training at that point (it is currently in
+progress), it was extremely beneficial for me to come as a helper to the first ever
+[Software Carpentry workshop in St Andrews](https://lmwake.github.io/2015-06-18-StAndrews/)
+in June 2015, and obtain an insight into the Software Carpentry teaching
+methodology.
+
+I took inspiration from the core Software Carpentry lessons,
+in particular from those on UNIX shell, Python and R.
+All of them have a central story which goes through almost every episode.
+For the GAP lesson, I have imagined a common situation: a
+research student with no prior experience of working with GAP (and perhaps
+little or no experience with programming at all) is facing a task to find
+a way in the huge library of GAP functions in order to study some research
+problem. Along this way, they start to work with GAP command line to explore
+algebraic objects interactively; then use the GAP language to write
+some simple scripts; then create own functions. More advanced topics
+such as, for example, extending GAP with new methods for existing types of
+objects, or even new objects, or organising your code in the form of a GAP
+package, are not so obvious for the beginners, and I have made an attempt
+to create a lesson which will show the direction in which their skills should
+be developing, and also to cover the importance of testing their code.
+
+I started from picking up a research-like problem which may nicely expose
+all needed techniques and explain the mindset required to deal with it.
+A good candidate was the problem of calculating an average order of an element
+of the group, which once I've seen being used by Steve Linton to quickly
+demonstrate some GAP features to a general scientific audience. I have tried to
+expand this problem in my talk in Newcastle in May 2015 (see the blog post
+[here](http://www.codima.ac.uk/2015/07/01/average-order-of-group-elements-a-demo-of-test-driven-development-in-gap/)),
+an thus the choice has been made.
+
+Indeed, the problem of determining an average order of the element of the group
```

Comment:
I think the following four paragraphs are more detail than our readers will want - can they be compressed to a single paragraph with pointers to the original (longer) blog post?